spring-ai-core/src/main/java/org/springframework/ai/embedding/EmbeddingResponseMetadata.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.embedding;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.ai.model.ResponseMetadata;
+
+/**
+ * @author Christian Tzolov
+ */
+public class EmbeddingResponseMetadata extends HashMap<String, Object> implements ResponseMetadata {
```

Comment:
maybe composition instead?  more for future proofing.  put this  list together of responses from the different providers.



OpenAI

metadata.put("model", model);
metadata.put("prompt-tokens", usage.promptTokens());
metadata.put("completion-tokens", usage.completionTokens());
metadata.put("total-tokens", usage.totalTokens());


Azure OpenAI


metadata.put("model", model);
metadata.put("prompt-tokens", embeddingsUsage.getPromptTokens());
metadata.put("total-tokens", embeddingsUsage.getTotalTokens());


PostgresML

Map.of(
"transformer", this.transformer,
"vector-type", this.vectorType.name(), 
"kwargs", this.kwargs));


Vertex

{
  "predictions": [
    {
      "embeddings": {
        "statistics": {
          "truncated": false,
          "token_count": 6
        },
        "values": [ ... ]
      }
    }
  ]
}